src/main/java/org/spongepowered/api/Server.java

```diff
@@ -86,6 +89,21 @@
     Optional<Player> getPlayer(String name);
 
     /**
+     * Gets a {@link Player} by their name, as seen by the
+     * provided viewer.
+     *
+     * <p>This only works for online players.</p>
+     *
+     * <p><b>Note: Do not use names for persistent storage, the
+     * Notch of today may not be the Notch of yesterday.</b></p>
+     *
+     * @param name The name to get the player from
+     * @param viewer The viewer of the request
+     * @return {@link Player} or Optional.empty() if not found
+     */
+    Optional<Player> getPlayer(String name, @Nullable ContextViewer viewer);
```

Comment:
Why is the viewer nullable? If you want a player as seen by no view then the normal methods are what you want.

Also why is this returning a Player and not a DataContext? is the player some facade for the DataContext associated with the given viewer?